File: server-lite/src/model/vacancy.py
from sqlalchemy import Column, Integer, String, Text, Boolean, DateTime, ForeignKey, Enum
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import UUID, ARRAY
from database import Base
import enum

class Vacancy(Base):
    __tablename__ = "vacancies"
    
    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(200), nullable=False)
    description = Column(Text, nullable=False)
    tags = Column(ARRAY(String))
    salary_from = Column(Integer, nullable=False)
    salary_to = Column(Integer, nullable=False)
    salary_currency = Column(String(3), default="RUB", nullable=False)
    location = Column(String(100), nullable=False)
    is_remote = Column(Boolean, default=False)
    is_active = Column(Boolean, default=True)
    
    employer_id = Column(UUID, ForeignKey("users.id"), nullable=False)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    employer = relationship("User", back_populates="vacancies")
    responses = relationship("Response", back_populates="vacancy")

class ResponseStatus(str, enum.Enum):
    PENDING = "pending"
    REJECTED = "rejected"
    ACCEPTED = "accepted"

class Response(Base):
    __tablename__ = "responses"
    
    id = Column(Integer, primary_key=True, index=True)
    status = Column(Enum(ResponseStatus), default=ResponseStatus.PENDING)
    
    vacancy_id = Column(Integer, ForeignKey("vacancies.id"), nullable=False)
    applicant_id = Column(UUID, ForeignKey("users.id"), nullable=False)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    vacancy = relationship("Vacancy", back_populates="responses")
    applicant = relationship("User", back_populates="responses")

File: app/EmploymentApp/Models/Vacancy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EmploymentApp.Models
{
    public class Vacancy
    {
        public int Id { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public List<string> Tags { get; set; } = new();

        public int SalaryFrom { get; set; }

        public int SalaryTo { get; set; }

        public string SalaryCurrency { get; set; } = "RUB";

        public string Location { get; set; }

        public bool IsRemote { get; set; }

        public bool IsActive { get; set; } = true;

        public int EmployerId { get; set; }

        // Вычисляемое свойство для отображения диапазона зарплаты
        public string SalaryRange =>
            $"{SalaryFrom:N0} — {SalaryTo:N0} {SalaryCurrency}";

        // Вычисляемое свойство для места работы
        public string WorkLocation =>
            IsRemote ? "Удаленно" : Location;

        public Vacancy(int id, string title, string description, 
            int salaryFrom, int salaryTo, 
            string location, 
            bool isRemote,
            int employerId, 
            List<string> tags = null, 
            string salaryCurrency = "RUB", 
            bool isActive = true)
        {
            Id = id;
            Title = title;
            Description = description;
            Tags = tags ?? new List<string>();
            SalaryFrom = salaryFrom;
            SalaryTo = salaryTo;
            SalaryCurrency = salaryCurrency;
            Location = location;
            IsRemote = isRemote;
            IsActive = isActive;
            EmployerId = employerId;
        }
    }
}
